### PR TITLE
Clean up linux sigprocmask, raise

### LINKED ISSUE #3903: sigprocmask (std.os.linux) questions
Currenty [std.os.linux](https://github.com/ziglang/zig/blob/master/lib/std/os/linux.zig) has
```zig
pub fn sigprocmask(flags: u32, noalias set: *const sigset_t, noalias oldset: ?*sigset_t) usize;
pub fn blockAllSignals(set: *sigset_t) void;
pub fn blockAppSignals(set: *sigset_t) void;
pub fn restoreSignals(set: *sigset_t) void;
```
shouldn't it be
```zig
pub fn sigprocmask(flags: u32, noalias set: ?*const sigset_t, noalias oldset: ?*sigset_t) isize;
pub fn blockAllSignals(set: ?*sigset_t) isize;
pub fn blockAppSignals(set: ?*sigset_t) isize;
pub fn restoreSignals(set: ?*sigset_t) isize;
```
That is,
- Return `isize` (or maybe `i32`), kernel can return `-EFAULT` (`-14`, bad address) and `-EINVAL` (`-22`, invalid argument). In POSIX it is `-1` with `errno` set (for `EINVAL`).
- `set` and `oset` can be `NULL` to opt-out from changing/querying. Both in Linux and POSIX.

### PR DESCRIPTION
**Pull request by daurnimator**
Closes #3903

### CODE REVIEW

File: lib/std/os/bits/linux.zig [hunk 1]
```diff
@@ -764,19 +764,14 @@ pub const winsize = extern struct {
     ws_ypixel: u16,
 };
 
+/// NSIG is the total number of signals defined.
+/// As signal numbers are sequential, NSIG is one greater than the largest defined signal number.
 pub const NSIG = if (is_mips) 128 else 65;
-pub const sigset_t = [128 / @sizeOf(usize)]usize;
```
Comment (daurnimator) [inline lib/std/os/bits/linux.zig]: I considered making this a `std.PackedIntArray(u1, 1024)`. Let me know if you'd like that.

File: lib/std/os/bits/linux.zig [hunk 2]
```diff
@@ -764,19 +764,14 @@ pub const winsize = extern struct {
...
-    }
-else
-    struct {
-        pub const all_mask = [4]u32{ 0xffffffff, 0xffffffff, 0xffffffff, 0xffffffff };
-        pub const app_mask = [4]u32{ 0xfffffffc, 0x7fffffff, 0xffffffff, 0xffffffff };
-    };
+pub const sigset_t = [1024/8]u8;
+
+pub const all_mask: sigset_t = [_]u8{ 0xff } ** sigset_t.len;
+pub const app_mask: sigset_t = [8]u8{ 0xff, 0xff, 0xff, 0xfc, 0x7f, 0xff, 0xff, 0xff } ++ [_]u8{ 0xff } ** 120;
```
Comment (LemonBoy) [inline lib/std/os/bits/linux.zig]: Mind the endianness.

Comment (daurnimator) [reply lib/std/os/bits/linux.zig]: Good call! Fixed.

**Comment by andrewrk** [comment]
Thanks! Really nice cleanup.

**Pull request merged**